### PR TITLE
Ephemeral support

### PR DESCRIPTION
**Pull request by kfox1111**
This adds support to the hostPath driver to be used
in an ephemeral mode.

### CODE REVIEW

**Comment by kfox1111** [comment]
@pohly Driver modified as suggested. I'll work on adding a deployment/CR, but the CR will need a flag marking it as ephemeral that doesn't yet exist in the api.  After digging through the code, it was not too bad to mix the use cases after all.

Tested with the csc tool. Seems to work properly.

@vladimirvivien @sbezverk

**Comment by kfox1111** [comment]
@vladimirvivien please take a look. This would be useful to have in for testing the pod inline code.

**Comment by vladimirvivien** [comment]
@kfox1111 I am testing this right now. I found couple places where it's throwing errors. I am going to submit a PR agains this PR so we can keep moving.

**Comment by kfox1111** [comment]
Sounds good. Thanks for testing/fixes.

**Comment by kfox1111** [comment]
@vladimirvivien Had to fix some merge conflicts. I think I got everything. Please take a look.

File: pkg/hostpath/nodeserver.go [hunk 1]
```diff
@@ -151,8 +154,22 @@ func (ns *nodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublis
 		}
 		mounter := mount.New("")
 		path := provisionRoot + volumeId
```
Comment (vladimirvivien) [inline pkg/hostpath/nodeserver.go]: Change to `path := getVolumePath(volumeId)` for consistency.

File: pkg/hostpath/controllerserver.go [hunk 2]
```diff
@@ -136,7 +139,7 @@ func (cs *controllerServer) CreateVolume(ctx context.Context, req *csi.CreateVol
 	}
 
 	volumeID := uuid.NewUUID().String()
-	path := provisionRoot + volumeID
+	path := getVolumePath(volumeID)
 
 	switch requestedAccessType {
 	case blockAccess:
```
Comment (vladimirvivien) [inline pkg/hostpath/controllerserver.go]: In line 234 (https://github.com/kubernetes-csi/csi-driver-host-path/blob/b8652fe0474d4788f7110d5f2a7bc4e3d082897b/pkg/hostpath/controllerserver.go#L234) 

change: `volPathHandler.GetLoopDevice(provisionRoot + vol.VolID)` ======> `volPathHandler.GetLoopDevice(getVolumePath(vol.VolID))` to remain consistent with how path is built.

**Comment by vladimirvivien** [comment]
@kfox1111 I submitted PR - https://github.com/kfox1111/csi-driver-host-path/pull/2 against your branch to fix the issues mentioned above and some other breaking issues I found while testing. PTAL.

**Comment by vladimirvivien** [comment]
@kfox1111 can you please rebase.

**Comment by kfox1111** [comment]
Merge conflict resolved and pr flattened.

**Comment by kfox1111** [comment]
Anything else to do?

**Comment by kfox1111** [comment]
looks like that worked. I'll squash it so we can merge.

**Comment by jsafrane** [comment]
Can't spot anything wrong
/approve

I let second part of eyes to lgtm

**Pull request merged**